packages/neon_framework/lib/src/utils/request_manager.dart

```diff
@@ -13,26 +11,54 @@ import 'package:rxdart/rxdart.dart';
 import 'package:sqflite/sqflite.dart';
 import 'package:xml/xml.dart' as xml;
 
+/// A callback that unwraps elements of type [R] into [T].
+///
+/// This is commonly used to get the relevant information from a broader response.
 typedef UnwrapCallback<T, R> = T Function(R);
+
+/// A callback to serialize a value of type [T] into a string.
+///
+/// This is used to store a value in the cache.
+@internal
 typedef SerializeCallback<T> = String Function(T);
+
+/// A callback to revive cached values [String] into their original type [T].
+@internal
 typedef DeserializeCallback<T> = T Function(String);
-typedef NextcloudApiCallback<T> = AsyncValueGetter<T>;
 
-const maxRetries = 3;
-const defaultTimeout = Duration(seconds: 30);
+/// A callback that retrieves a value of type [R] from the Nextcloud api.
+typedef NextcloudApiCallback<R> = AsyncValueGetter<R>;
 
+/// The count of how many times to retry failing requests.
```

Comment:
Technically this description is wrong because the number indicates how many tries will be made, not how many retries (which would be +1 for the original request).